include/ddwaf.h

```diff
@@ -477,6 +477,101 @@ bool ddwaf_object_map_addl(ddwaf_object *map, const char *key, size_t length, dd
  **/
 bool ddwaf_object_map_addl_nc(ddwaf_object *map, const char *key, size_t length, ddwaf_object *object);
 
+/**
+ * ddwaf_object_type
+ *
+ * Returns the type of the object.
+ *
+ * @param object The object from which to get the type.
+ *
+ * @return The object type of DDWAF_OBJ_INVALID if NULL.
+ **/
+DDWAF_OBJ_TYPE ddwaf_object_type(ddwaf_object *object);
```

Comment:
I'd have preferred definitions with `inline`. You could then add a .c file compiled with `-std=c11` where you just had `extern inline DDWAF_OBJ_TYPE ddwaf_object_type();`. This should work for both C and C++ users of the header.